src/components/SelectableMainTable.tsx

```diff
@@ -76,41 +89,56 @@ const SelectableMainTable: FC<Props> = ({
             checked={isAllSelected}
             indeterminate={isSomeSelected && !isAllSelected}
             onChange={isSomeSelected ? selectNone : selectPage}
+            disabled={disableSelect}
           />
-          <ContextualMenu
-            className="select-context-menu"
-            position="left"
-            title="Multiselect"
-            toggleAppearance="base"
-            toggleClassName="has-icon u-no-margin--bottom"
-            toggleLabel={<Icon name="chevron-down" />}
-            toggleProps={{
-              "aria-label": "multiselect rows",
-            }}
-            links={[
-              {
-                children: `Select all ${itemName}s on this page`,
-                onClick: selectPage,
-              },
-              {
-                children: `Select all ${parentName} ${itemName}s`,
-                onClick: selectAll,
-              },
-            ]}
-          />
+          {!hideContextualMenu && (
+            <ContextualMenu
+              className="select-context-menu"
+              position="left"
+              title="Multiselect"
+              toggleAppearance="base"
+              toggleClassName="has-icon u-no-margin--bottom"
+              toggleLabel={<Icon name="chevron-down" />}
+              toggleProps={{
+                "aria-label": "multiselect rows",
+                disabled: disableSelect,
+              }}
+              links={[
+                {
+                  children: `Select all ${pluralize(itemName, 2)} on this page`,
+                  onClick: selectPage,
+                },
+                {
+                  children: `Select all ${parentName} ${pluralize(itemName, 2)}`,
+                  onClick: selectAll,
+                },
+              ]}
+            />
+          )}
         </>
       ),
-      className: "select select-header",
+      className: classnames("select select-header", {
+        "no-menu": hideContextualMenu,
+      }),
       "aria-label": "select",
     },
     ...(headers ?? []),
   ];
 
+  const selectedNamesLookup = new Set(selectedNames);
+  const processingNamesLookup = new Set(processingNames);
+  const indeterminateNamesLookup = new Set(indeterminateNames);
   const rowsWithCheckbox = rows.map((row, rowIndex) => {
-    const isRowSelected = selectedNames.includes(row.name ?? "");
-    const isRowProcessing = processingNames.includes(row.name ?? "");
+    const isRowSelected = selectedNamesLookup.has(row.name ?? "");
+    const isRowProcessing = processingNamesLookup.has(row.name ?? "");
+    const isRowIndeterminate = indeterminateNamesLookup.has(row.name ?? "");
 
     const toggleRow = (event: PointerEvent<HTMLInputElement>) => {
+      if (onToggleRow) {
+        onToggleRow(row, selectedNames);
```

Comment:
I think we are only ever using the `row.name`, and it is probably not good to leak the whole row structure to a parent.
```suggestion
        onToggleRow(row.name);
```